lib/ansible/modules/monitoring/icinga2_host.py

```diff
@@ -133,14 +133,18 @@
 #
 class icinga2_api:
     module = None
+    use_proxy = True
+
+    def __init__(self, use_proxy=True):
+        self.use_proxy = use_proxy
 
     def call_url(self, path, data='', method='GET'):
         headers = {
             'Accept': 'application/json',
             'X-HTTP-Method-Override': method,
         }
         url = self.module.params.get("url") + "/" + path
-        rsp, info = fetch_url(module=self.module, url=url, data=data, headers=headers, method=method)
+        rsp, info = fetch_url(module=self.module, url=url, data=data, headers=headers, method=method, use_proxy=self.use_proxy)
```

Comment:
I think it's better to use `use_proxy=module.params['use_proxy']` here (and avoid the `use_proxy` property).

(In my opinion, `fetch_url()` should use `use_proxy` from `module.params` if not explicitly specified, but that's probably too late to change, or at least too much effort. Moving the source of `use_proxy` as close as possible to the `fetch_url` call is at least making it easier to correct this mistake eventually in the future.)

Reply:
@felixfontein could you check if you're OK with current approach?

Reply:
Yes, it's good this way!